hikari/polls.py

```diff
@@ -0,0 +1,325 @@
+        self._allow_multiselect = allow_multiselect
+        self._layout_type = layout_type
+
+        # Answer is required, but we want users to user add_answer() instead of
+        # providing at initialization.
+        #
+        # Considering that answer ID can be arbitrary, `list`-based approaches
+        # like that of hikari.embeds.Embed._fields, while feasible to implement,
+        # would decrease long-term maintainability. I'm opting to use a `dict`
+        # here to simplify the implementation with some performance trade-off
+        # due to hashmap overhead.
+        self._answers: typing.MutableMapping[int, PollAnswer] = {}  # TODO: Do we need to set to None?
+
+        # Counter to keep track of the answer IDs
+        #
+        # Discord (at the time of writing this) starts at 1, I'm opting to follow
+        # the behaviour.
+        self._counter = 1
+
+    @property
+    def question(self) -> PollMedia:
+        """Returns the question of the poll."""
+        return self._question
+
+    @question.setter
+    def question(self, value: str) -> None:
+        self._question = PollMedia(text=value)
+
+    @property
+    def allow_multiselect(self) -> bool:
+        """Returns whether the poll allows multiple answers."""
+        return self._allow_multiselect
+
+    @allow_multiselect.setter
+    def allow_multiselect(self, value: bool) -> None:
+        self._allow_multiselect = value
+
+    @property
+    def layout_type(self) -> PollLayoutType:
+        """Returns the layout type of the poll."""
+        return PollLayoutType(self._layout_type)
+
+    @layout_type.setter
+    def layout_type(self, value: typing.Union[int, PollLayoutType]) -> None:
+        self._layout_type = value
+
+    @property
+    def answers(self) -> typing.MutableMapping[int, PollAnswer]:
+        """Returns the answers of the poll.
+
+        !!! note
+            Use [`hikari.polls.Poll.add_answer`][] to add a new answer,
+            [`hikari.polls.Poll.edit_answer`][] to edit an existing answer, or
+            [`hikari.polls.Poll.remove_answer`][] to remove a answer.
+        """
+        return self._answers
+
+
+class PollCreate(PartialPoll):
+    """Used to create a poll."""  # TODO: Improve this docstring
+
+    __slots__: typing.Sequence[str] = ("_duration",)
+
+    def __init__(
+        self,
+        question: str,
+        duration: int,
+        allow_multiselect: bool,
+        layout_type: typing.Union[int, PollLayoutType] = PollLayoutType.DEFAULT,
+    ):
+
+        super().__init__(question=question, allow_multiselect=allow_multiselect, layout_type=layout_type)
+        self._duration = duration
+
+    @property
+    def duration(self) -> int:
+        """Returns the duration of the poll."""
+        return self._duration
+
+    @duration.setter
+    def duration(self, value: int) -> None:
+        self._duration = value
+
+    def add_answer(self, text: str, emoji: typing.Optional[Emoji]) -> PartialPoll:
```

Comment:
Allow me to set my own `answer_id`. Not sure how to actually implement it to make sure you don't have multiple of the same. Maybe raise some `UniqueError` of some kind?

Reply:
Merged and resolved